### PR TITLE
Network udp

### PR DESCRIPTION
**Pull request by bkolad**

### CODE REVIEW

File: network/udp/net.go [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+
+	conn, err := net.DialUDP("udp", nil, udpAddr)
+	if err != nil {
+		panic(err)
+	}
+	defer conn.Close()
+
+	byteWriter := bufio.NewWriter(conn)
+	// The packets are "gob" encoded
+	enc := gob.NewEncoder(byteWriter)
```
Comment (bkolad) [inline network/udp/net.go]: @nikkolasg here we are using gob format for packet serialization, the Marshal/Unmarshal machinery from handel package can't be reused as we need length encoding in order to know how many bytes to read from the network.  
Shall we use gob as a default serialization format?

File: network/makestaticregistry.go [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+			listenPort = pt
+		}
+		addr := fmt.Sprintf("%s:%d", rec[2], pt)
+
+		//TODO read peer's serialized public key
+		node := nodeRecord{id: id, port: pt, addr: addr, pubKey: ""}
+		var identity h.Identity = node
+		nodes = append(nodes, identity)
+	}
+	return h.NewArrayRegistry(nodes), listenPort, nil
```
Comment (bkolad) [inline network/makestaticregistry.go]: @nikkolasg NewArrayRegistry(ids) assumes ids slice is sorted.  Can we sort the array in the NewArrayRegistry func
to make it more convenient?

File: network/makestaticregistry.go [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+}
+
+// ReadCSV creates a Registry based on provided CSV file.
+// Returns the registry and listen port of the local peer/
+func ReadCSV(path string, localID int32) (h.Registry, int) {
```
[the remark this answers is not in the document]
Comment (nikkolasg) [reply network/makestaticregistry.go]: Probably 
```go
   Read(element []string) (Identity,error)
```

File: network/udp/net.go [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+
+// NewUDPNetwork creates Nework baked by udp protocol
+func NewUDPNetwork(listenPort int) *udpNet {
+	addr := fmt.Sprintf("0.0.0.0:%d", listenPort)
+	udpAddr, err := net.ResolveUDPAddr("udp4", addr)
+	if err != nil {
+		panic(err)
+	}
+
+	conn, err := net.ListenUDP("udp", udpAddr)
```
[the remark this answers is not in the document]
Comment (bkolad) [reply network/udp/net.go]: done

File: network/makestaticregistry.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (bkolad) [reply network/makestaticregistry.go]: I did 
```go 
Read(line []string) (*NodeRecord, error) 
```
as we need port of the local peer and NodeRecord implements Identity

File: network/udp/net.go [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+	"fmt"
+	"io"
+	"log"
+	"net"
+
+	h "github.com/ConsenSys/handel"
+)
+
+type udpNet struct {
+	listeners *[]h.Listener
```
[the remark this answers is not in the document]
Comment (bkolad) [reply network/udp/net.go]: Done

File: network/udp/net.go [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+	err = enc.Encode(packet)
+	if err != nil {
+		//TODO consider changing it to error logging
+		panic(err)
+	}
+	byteWriter.Flush()
+}
+
+func handler(listeners *[]h.Listener, conn *net.UDPConn) {
+	for {
```
[the remark this answers is not in the document]
Comment (bkolad) [reply network/udp/net.go]: Done

File: network/makestaticregistry.go (hunk 2 above)
Comment (bkolad) [reply network/makestaticregistry.go]: Will do it in separate PR

File: network/udp/net.go [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+func (udpNet *udpNet) handler() {
+	for {
+		//udpNet.quit and udpNet.listeners have to be guarded by a read lock
+		udpNet.RLock()
+		quit := udpNet.quit
+		listeners := udpNet.listeners
+		enc := udpNet.enc
+		udpNet.RUnlock()
+
+		if quit {
```
Comment (nikkolasg) [inline network/udp/net.go]: Should that not be inside the Lock ?

**Pull request merged**